homeassistant/components/hifiberry/media_player.py

```diff
@@ -0,0 +1,222 @@
+
+    @property
+    def media_track(self):
+        """Track number of current playing media, music track only."""
+        return self._metadata.get("tracknumber", None)
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        art_url = self._metadata.get("artUrl", None)
+        external_art_url = self._metadata.get("externalArtUrl", None)
+        if art_url is not None:
+            if art_url.startswith("static/"):
+                return external_art_url
+            if art_url.startswith("artwork/"):
+                return f"{self._audiocontrol2.base_url}/{art_url}"
+            return art_url
+        return external_art_url
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return int(self._volume) / 100
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._muted
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._metadata.get("playerName")
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_HIFIBERRY
+
+    async def async_media_next_track(self):
+        """Send media_next command to media player."""
+        await self._audiocontrol2.player("next")
+
+    async def async_media_previous_track(self):
+        """Send media_previous command to media player."""
+        await self._audiocontrol2.player("previous")
+
+    async def async_media_play(self):
+        """Send media_play command to media player."""
+        await self._audiocontrol2.player("play")
+
+    async def async_media_pause(self):
+        """Send media_pause command to media player."""
+        await self._audiocontrol2.player("pause")
+
+    async def async_volume_up(self):
+        """Service to send the hifiberry the command for volume up."""
+        await self._audiocontrol2.volume("+5")
+        self._volume += 5
+
+    async def async_volume_down(self):
+        """Service to send the hifiberry the command for volume down."""
+        await self._audiocontrol2.volume("-5")
+        self._volume -= 5
+
+    async def async_set_volume_level(self, volume):
+        """Send volume_set command to media player."""
+        if volume < 0:
+            volume = 0
+        elif volume > 1:
+            volume = 1
+        await self._audiocontrol2.volume(int(volume * 100))
+        self._volume = volume * 100
+
+    async def async_mute_volume(self, mute):
+        """Mute. Emulated with set_volume_level."""
+        if mute:
+            self._muted_volume = self.volume_level
+            await self._audiocontrol2.volume(0)
+        await self._audiocontrol2.volume(int(self._muted_volume * 100))
```

Comment:
Should this only be called when muted is false? It seems like its undoing the mute in the mute case.